update docker configs

parkervcp opened the pull request:
Dockerfile:
updated start command
aligns the config file location 


docker-compose.example.yml
file adds custom network name so it can be used in firewalld commands.
updates volumes

parkervcp committed:
update docker configs

dockerfile with an updated start command

docker-compose file adds custom network name so it can be used in firewalld commands.

DaneEveritt commented on docker-compose.example.yml:
```diff
@@ -13,14 +14,20 @@ services:
-      - "/var/lib/pterodactyl/:/var/lib/pterodactyl/"
+      - "/etc/pterodactyl/:/etc/pterodactyl/"
+      - "/var/log/pterodactyl/:/var/log/pterodactyl/"
       - "/srv/daemon-data/:/srv/daemon-data/"
```
`/srv/daemon-data` is deprecated and only used by old nodes now. `/var/lib/pterodactyl/voulumes` replaces it.

DaneEveritt commented on docker-compose.example.yml:
```diff
@@ -13,14 +14,20 @@ services:
       - "/var/lib/docker/containers/:/var/lib/docker/containers/"
-      - "/var/lib/pterodactyl/:/var/lib/pterodactyl/"
+      - "/etc/pterodactyl/:/etc/pterodactyl/"
+      - "/var/log/pterodactyl/:/var/log/pterodactyl/"
```
I feel like all of these pterodactyl specific mounts should live in a single root directory for the instance, rather than scattered all over the host filesystem. That's how I've seen many other docker setups work. e.g. `~/wings/[logs|lib|etc.]`. I feel like scattering them all over, especially given how many things get mounted, is counter to the point of docker.

parkervcp replied:
linux standard is to put them where they are now.

I could add symlinks in the container but I have seen those fail.

parkervcp replied:
Also the daemon expects some of these to be in specific places on the host as well due to the bind mounts for the game server containers.

parkervcp replied:
@DaneEveritt Do you have a recommendation for where I "should" put these?

DaneEveritt replied:
Some type of Pterodactyl specific data directory. I mean, `/etc/pterodactyl/[etc,lib,log]` could work, or whatever the most logical place to put all of the stuff for a given program on the OS would be.

parkervcp replied:
We cant really consolidate these. I have the volumes and the reason why.

/var/run/docker.sock - Docker control
/var/lib/docker/containers/ - Docker logs
/etc/pterodactyl/ - wings config
/var/lib/pterodactyl/ - pterodactyl servers
/var/log/pterodactyl/ - pterodactyl logging (standard location, could probably move.)
/tmp/pterodactyl/ - installl scripts 

/srv/daemon-data/ - updated from old wings installl
/etc/letsencrypt/ - easier for letsencrypt automation for certs.

DaneEveritt replied:
```
 /etc/pterodactyl/ - wings config
/var/lib/pterodactyl/ - pterodactyl servers
/var/log/pterodactyl/ - pterodactyl logging (standard location, could probably move.)
/tmp/pterodactyl/ - installl scripts
```

All four of these can safely be mounted as `/etc/pterodactyl/[etc,lib,log,tmp]:[respective in-container path]`

DaneEveritt commented on docker-compose.example.yml:
```diff
@@ -13,14 +14,20 @@ services:
     volumes:
```
Please also pass `TZ=`

parkervcp committed:
update compose file

mount changes
remove /srv/daemon-data
remove /etc/timezone

variable changes
add TZ

parkervcp committed:
add note about old data folder.

matthewpi commented:
Could we also update the `Dockerfile` to use `1.15-alpine` instead of `1.14-alpine`?

parkervcp committed:
update to go 1.15

Update base image to go version 1.15
